### PR TITLE
Create config subpackage

### PR DESCRIPTION
**Pull request by leotrs**
### Context
This PR belongs to the ongoing refactor of the config system (#337).

### Main change
The main goal here is to encapsulate everything having to do with setting up the config in a new folder `config/`. Before, setting up the config and parsing the .cfg files was strewn about in many files. Now... it is still fairly disordered and in many files, but at least they live in the same directory. The next PR in this series will start merging those files, now within the safety of the new subpackage.

Yes, there are 30 files touched by this PR but most of them are one-line modifications of import statements. The actual PR is like two files and a bunch of renamed/moved files.

### The motivation
The main benefit of this change is that after the config is set up, no other part of the codebase needs to care about how it is being setup. Accordingly, many imports of the form `from ..config import camera_config` or similar have been changed to `from .. import camera_config`. The former says "go to the config module and get me the configuration of the camera", and it required of the developer that they know all of the internals of how the config system is set up. For example, if the configuration system changed (say is `camera_config` was defined in some other module), all of those imports had to change. The new version `from .. import camera_config` says "go to the top level package and get me the configuration of the camera". Even if the configuration system changes internally, as long as the config subpackage exports `camera_config` to the top-level package, this import will continue to work always.

Additionally, the `logger.py` module is also now part of the `config/` folder, as it deals with parsing .cfg files and needs to export a variable that all other modules need as soon as possible (the `logger`).

### List of changes
1. __init__.py now just imports `*` from the config subpackage.
2. There is a new folder `config/` which is made into a subpackage by adding the file `config/__init__.py`. Remember, a subpackage is nothing more than a collection of modules and the `config/__init__.py` file just determines what is exported into the top level package. The new subpackage contains all of the modules and resources related to setting up the config system.
3. A bunch of imports across the codebase have been simplified.
4. Added a couple of tests to check that everything is being exported adequately.

### Future work
Encapsulating everything inside a subpackage will make it super easy in the future to keep refactoring the config system with minimal modification of the rest of the codebase. See #337 for future plans on refactoring the config system.

EDIT: sorry for the messy commit history. I'm... trying something.

### CODE REVIEW

File: tests/test_imports.py [hunk 1]
```diff
@@ -1,2 +1,13 @@
+from types import ModuleType
+
 def test_import():
```
Comment (huguesdevimeux) [inline tests/test_imports.py:3]: Praise : nice test

**Review by huguesdevimeux — APPROVED** [review]

**Review by behackl — APPROVED** [review]
LGTM, nice cleanup.

**Comment by leotrs** [comment]
Ok _I think_ I know what's wrong with the tests. This PR moves the file `default.cfg` and so it is no longer being shipped with the rest of the package so when GH actions installs manim using pip, it can't find it. I can fix the setup.py in a sec.

@naveen521kk can we make sure this doesn't happen with poetry? The `default.cfg` file must always be shipped/packaged.

**Comment by naveen521kk** [comment]
> can we make sure this doesn't happen with poetry? The default.cfg file must always be shipped/packaged.

Yes, I had included everything in manim folder inside package.

**Review by Aathish04 — APPROVED** [review]

**Pull request merged**